modules/azure/vnet/nic-bastion.tf

```diff
@@ -9,12 +9,12 @@ resource "azurerm_network_interface" "bastion" {
     private_ip_address_allocation = "dynamic"
     name                          = "${var.cluster_name}-bastionIPConfiguration"
     subnet_id                     = "${element(azurerm_subnet.bastion_subnet.*.id, count.index)}"
-    public_ip_address_id          = "${element(azurerm_public_ip.bastion_public_ip.*.id, count.index)}"
+    public_ip_address_id          = "${var.vpn_enabled ? "" : element(concat(azurerm_public_ip.bastion_public_ip.*.id, list("")), count.index)}"
```

Comment:
this `concat` `list` is just workaround, becase element does not support empty list.